n9kv/docker/launch.py

```diff
@@ -0,0 +1,186 @@
+        self.qemu_args.extend(["-boot", "c"])
+        replace_index = self.qemu_args.index(
+            "if=ide,file={}".format(overlay_disk_image)
+        )
+        self.qemu_args[
+            replace_index
+        ] = "file={},if=none,id=drive-sata-disk0,format=qcow2".format(
+            overlay_disk_image
+        )
+        self.qemu_args.extend(["-device", "ahci,id=ahci0,bus=pci.0"])
+        self.qemu_args.extend(
+            [
+                "-device",
+                "ide-hd,drive=drive-sata-disk0,bus=ahci0.0,id=drive-sata-disk0,bootindex=1",
+            ]
+        )
+
+    def bootstrap_spin(self):
+        """This function should be called periodically to do work."""
+        if self.spins > 300:
+            # too many spins with no result ->  give up
+            self.stop()
+            self.start()
+            return
+
+        (ridx, match, res) = self.tn.expect([b"\(yes\/skip\/no\)\[no\]:", b"login:"], 1)
+        if match:  # got a match!
+            if ridx == 0:
+                self.logger.debug("matched poap prompt")
+                self.wait_write("yes", wait=None)
+                self.wait_write(
+                    "no", wait="Do you want to enforce secure password standard"
+                )
+                self.wait_write(self.password, wait='Enter the password for "admin"')
+                self.wait_write(self.password, wait='Confirm the password for "admin"')
+                self.wait_write(
+                    "no", wait="Would you like to enter the basic configuration dialog"
+                )
+            elif ridx == 1:  # login
+                self.logger.debug("matched login prompt")
+                self.logger.debug(f'trying to log in with "admin" / {self.password}')
+                self.wait_write("admin", wait=None)
+                self.wait_write(self.password, wait="Password:")
+
+                # run main config!
+                self.bootstrap_config()
+                # close telnet connection
+                self.tn.close()
+                # startup time?
+                startup_time = datetime.datetime.now() - self.start_time
+                self.logger.info("Startup complete in: %s" % startup_time)
+                # mark as running
+                self.running = True
+                return
+
+        # no match, if we saw some output from the router it's probably
+        # booting, so let's give it some more time
+        if res != b"":
+            self.logger.trace("OUTPUT: %s" % res.decode())
+            # reset spins if we saw some output
+            self.spins = 0
+
+        self.spins += 1
+
+        return
+
+    def bootstrap_config(self):
+        """Do the actual bootstrap config"""
+        self.logger.info("applying bootstrap configuration")
+        self.wait_write("", None)
+        self.wait_write("configure")
+        self.wait_write(
+            f"username {self.username} password 0 {self.password} role network-admin"
+        )
+
```

Comment:
I don't see if you have `hostname` provisioned anywhere?

Reply:
fixed